govcd/vdc_template.go

```diff
@@ -0,0 +1,238 @@
+		}
+		return vcdClient.GetVdcTemplateById(results.Results.AdminOrgVdcTemplateRecord[0].HREF)
+	} else {
+		if len(results.Results.OrgVdcTemplateRecord) == 0 {
+			return nil, fmt.Errorf("could not find any VDC Template with name '%s': %s", name, ErrorEntityNotFound)
+		}
+		if len(results.Results.OrgVdcTemplateRecord) > 1 {
+			return nil, fmt.Errorf("expected one VDC Template with name '%s', but got %d", name, len(results.Results.OrgVdcTemplateRecord))
+		}
+		return vcdClient.GetVdcTemplateById(results.Results.OrgVdcTemplateRecord[0].HREF)
+	}
+}
+
+// Delete deletes the receiver VDC Template
+func (vdcTemplate *VdcTemplate) Delete() error {
+	if !vdcTemplate.client.IsSysAdmin {
+		return fmt.Errorf("functionality requires System Administrator privileges")
+	}
+	if vdcTemplate.VdcTemplate.HREF == "" {
+		return fmt.Errorf("cannot delete the VDC Template, its HREF is empty")
+	}
+
+	_, err := vdcTemplate.client.ExecuteRequest(vdcTemplate.VdcTemplate.HREF, http.MethodDelete, "", "error deleting VDC Template: %s", nil, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetAccess sets the Access control configuration for the receiver VDC Template,
+// which specifies which Organizations can read it.
+func (vdcTemplate *VdcTemplate) SetAccess(organizationIds []string) error {
+	if !vdcTemplate.client.IsSysAdmin {
+		return fmt.Errorf("functionality requires System Administrator privileges")
+	}
+	if vdcTemplate.VdcTemplate.HREF == "" {
+		return fmt.Errorf("cannot set the Access list for the VDC Template, its HREF is empty")
+	}
+	accessSettings := make([]*types.AccessSetting, len(organizationIds))
+	for i, organizationId := range organizationIds {
+		accessSettings[i] = &types.AccessSetting{
+			Subject: &types.LocalSubject{
+				HREF: fmt.Sprintf("%s/org/%s", vdcTemplate.client.VCDHREF.String(), extractUuid(organizationId))},
+			AccessLevel: types.ControlAccessReadOnly,
+		}
+	}
+	payload := &types.ControlAccessParams{AccessSettings: &types.AccessSettingList{AccessSetting: accessSettings}}
+
+	return vdcTemplate.client.setAccessControlWithHttpMethod(http.MethodPut, payload, vdcTemplate.VdcTemplate.HREF, "VDC Template", vdcTemplate.VdcTemplate.Name, nil)
+}
+
+// GetAccess retrieves the Control access configuration for the receiver VDC Template, which
+// contains the Organizations that can read it.
+func (vdcTemplate *VdcTemplate) GetAccess() (*types.ControlAccessParams, error) {
+	if !vdcTemplate.client.IsSysAdmin {
+		return nil, fmt.Errorf("functionality requires System Administrator privileges")
+	}
+	if vdcTemplate.VdcTemplate.HREF == "" {
+		return nil, fmt.Errorf("cannot get the Access list for the VDC Template, its HREF is empty")
+	}
+	result := &types.ControlAccessParams{}
+	href := fmt.Sprintf("%s/controlAccess", vdcTemplate.VdcTemplate.HREF)
+	_, err := vdcTemplate.client.ExecuteRequest(href, http.MethodGet, types.AnyXMLMime, "error getting access of VDC Template: %s", nil, result)
```

Comment:
> error getting access of VDC Template

This reads strange. Is there a mistake in phrasing this sentence?

Reply:
Yeah, I think that log entry and the methods needed a better naming.

Changed to `Access control` and `AccessControl` instead of just `Access` everywhere.